components/org.wso2.carbon.identity.api.user.push.device.handler.v1/src/main/resources/push-auth.yaml

```diff
@@ -0,0 +1,780 @@
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        403:
+          description: Forbidden
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        500:
+          description: Server Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+      tags:
+        - me
+
+    put:
+      description: |
+        This API is used to update attributes of a registered device.<br/>
+        <b>Permission required:</b>
+         * /permission/admin/login <br/>
+          <b>OAuth2 Scopes:</b>
+         * internal_identity_mgt_update
+      summary: |
+        Update attributes of a registered device.
+      requestBody:
+        description: Optional description in *Markdown*
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/PatchDTO'
+      parameters:
+        - name: deviceId
+          in: path
+          description: deviceId
+          required: true
+          schema:
+            type: string
+      responses:
+        200:
+          description: Display name of device successfully updated
+        400:
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        401:
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        403:
+          description: Forbidden
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        404:
+          description: Not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        500:
+          description: Server Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+      tags:
+        - me
+
+    get:
+      description: |
+        This API is used to get a specific device.<br/>
+        <b>Permission required:</b>
+         * /permission/admin/login <br/>
+          <b>OAuth2 Scopes:</b>
+         * internal_identity_mgt_view
+      summary: |
+        Returns specific device.
+      parameters:
+        - name: "deviceId"
+          in: "path"
+          description: "ID of device to return"
+          required: true
+          schema:
+            type: "string"
+      responses:
+        200:
+          description: Details of a specific device
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DeviceDTO'
+        400:
```

Comment:
What can cause a 400 response here?

Reply:
Replaced with 404 - Not found as the 400 was thrown when the device wasn't found